src/java/htsjdk/tribble/index/IndexFactory.java

```diff
@@ -420,8 +419,32 @@ private FeatureCodecHeader readHeader() {
 
         private PositionalBufferedStream initStream(final File inputFile, final long skip) {
             try {
-                final FileInputStream is = new FileInputStream(inputFile);
+                final FileInputStream fileStream = new FileInputStream(inputFile);
+                final InputStream is;
+
+                // if this looks like a block compressed file and it in fact is, we will use it
+                // otherwise we will use the file as is
+                if (AbstractFeatureReader.hasBlockCompressedExtension(inputFile)) {
+                    // make a buffered stream to test that this is in fact a valid block compressed file
+                    final int bufferSize = Math.max(Defaults.BUFFER_SIZE, BlockCompressedStreamConstants.MAX_COMPRESSED_BLOCK_SIZE);
+                    final BufferedInputStream bufferedStream = new BufferedInputStream(fileStream, bufferSize);
+
+                    if (!BlockCompressedInputStream.isValidFile(bufferedStream)) {
+                        throw new TribbleException.MalformedFeatureFile("Input file is not in valid block compressed format.", inputFile.getAbsolutePath());
+                    }
+
+                    // if we got here, the file is valid, make a SeekableStream for the BlockCompressedInputStream to read from
+                    final SeekableStream seekableStream =
+                            SeekableStreamFactory.getInstance().getBufferedStream(
```

Comment:
I would save a local reference to the `SeekableStreamFactory` instance here, to make it a little more readable.

But that's just personal preference, ignore if you disagree.